### PR TITLE
Bring improved error messages from requesting api 

### PR DESCRIPTION
**Pull request by melanierogan**
## Who is this for?
This is for people who will be requesting items from the collection. The requesting service API can respond with an error under several circumstances. At present we just tell the user that the request failed and ask then to try again. 
Ticket: https://github.com/wellcomecollection/wellcomecollection.org/issues/6916

## What is it doing for them?

We want to ensure we are displaying informative error messages to people using the requesting service.

## What does that look like?

I was able to replicate the error of trying to request an item when I have already reached my limit, this at least demonstrates the process of getting the `error.description` back from the API. In theory, it should work for each error case set by the API. 

requestingerrormessage_opt

## Please note
I saw [a PR](https://github.com/wellcomecollection/catalogue-api/pull/319) that has been merged in the API to change the error copy to use can't instead of cannot, but you'll see in the attached gif that cannot still appears. I'll need to investigate that.

### CODE REVIEW

File: catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx [hunk 1]
```diff
@@ -382,16 +386,18 @@ const ConfirmedDialog: FC<ConfirmedDialogProps> = ({ currentHoldNumber }) => (
...
 };
 
-const ErrorDialog: FC<ErrorDialogProps> = ({ setIsActive }) => (
+const ErrorDialog: FC<ErrorDialogProps> = ({ setIsActive, errorMessage }) => (
   <>
     <Header>
       <span className={`h2`}>Request failed</span>
     </Header>
     <p className="no-margin">
       {/* TODO: get error code and construct appropriate message from response - see #6916 */}
```
Comment (alexwlchan) [inline catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx:398]: Remember to remove this TODO line!

File: catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx [hunk 2]
```diff
@@ -382,16 +386,18 @@ const ConfirmedDialog: FC<ConfirmedDialogProps> = ({ currentHoldNumber }) => (
...
-      There was a problem requesting this item. Please try again.
+      {errorMessage ||
+        'There was a problem requesting this item. Please try again.'}
```
Comment (alexwlchan) [inline catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx:400]: While we're here, maybe we move this default error message to the microcopy file?

File: catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx [hunk 3]
```diff
@@ -442,9 +449,11 @@ const ItemRequestModal: FC<Props> = ({
           'Content-Type': 'application/json',
         },
       });
+      const responseJson = await response.json();
```
Comment (alexwlchan) [inline catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx:452]: nit: do we need to look up the response JSON if the request is okay?

File: catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx [hunk 4]
```diff
@@ -454,16 +463,27 @@ const ItemRequestModal: FC<Props> = ({
...
       // TODO: error to Sentry?
     }
   }
 
-  function renderModalContent(requestingState: RequestingState) {
+  function renderModalContent(
+    requestingState: RequestingState,
+    requestingErrorMessage?:
+      | string
+      | 'There was a problem requesting this item. Please try again.'
```
Comment (alexwlchan) [inline catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx:475]: Why are we duplicating the default error message here? Do we need it in this function signature?

**Review by alexwlchan — COMMENTED** [review]
🦾

File: catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx (hunk 4 above)
Comment (melanierogan) [reply catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx:475]: Ah yep, that was me messing about making typescript happy as I was trying things out - I'll adjust now!

File: catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx (hunk 2 above)
Comment (melanierogan) [reply catalogue/webapp/components/ItemRequestModal/ItemRequestModal.tsx:400]: Great shout!

**Review by jamieparkinson — APPROVED** [review]
✨

**Pull request merged**